### PR TITLE
[WIP] Allow nuke ops to start as their own species, add EMP shield

### PR DESCRIPTION
**Pull request by datlo**
This PR refactors nuke ops setup to fix issues with some species, and lets the op choose whether they want to spawn as their character's species or as a randomised human. 

The EMP shield implant is also added to the nuke ops uplink for 6 TC. "This implant will protect you from the effects of 5 EMPs or Ion shots, recharging once per 30 seconds."

TODO : Add specie specific sprite to the nukie elite hardsuit

🆑
add: Nuclear operatives can now choose to spawn as either their character or a randomised human.
fix: Fixed plasmaman operative burning on spawn. (they get healed once the round is done setting up)
Add: Added EMP Shield implant to the nuke ops uplink for 6 TC. Can block 5 EMPs, recharges once per 30 seconds.
/🆑

### CODE REVIEW

**Comment by Mithrandalf** [comment]
Something like this needs a design doc and a long think about the balance implications imo

It's a shame cos I almost like it

**Comment by FalseIncarnate** [comment]
This was deliberately removed after extensive discussion among the staff, for reasons including (but not limited to) species' weaknesses forcing specific equipment and play-style choices.

You are attempting to return this AND effectively force the same choices be made all over again.

Who among the staff have you discussed this with?

**Comment by datlo** [comment]
This was discussed with spartan and necalundun (and mb other staff i cant recall on the coder chat discord)

This PR does not force players to play as their set specie. Nuke ops get a prompt on spawn to human.

**Comment by FalseIncarnate** [comment]
I realize my previous comment may have come across as hostile or confrontational, which was not my intent. I am not outright opposed to this idea (variety is definitely a good thing when it comes to keeping game modes interesting), but wanted to make sure this was not being done lightly, without adequate consideration, or without consulting with staff first.

**Comment by datlo** [comment]
It's fine, it's hard to convey intent over text, I always give people the benefit of the doubt.

Do you have a link to the relevant PR or commit that changed nuke ops to be human only so I can see relevant discussion/code changes from then? As far as I can tell looking at the history nuke ops have always been human only but that's probably me sucking at browsing git

**Comment by Ty-Omaha** [comment]
Bad idea

Vox and IPC get to be damage modified +

Slime people will bleed water and slip teammates

Plasmamen don't get to have a syndicate space suit, or if they do have a special plasmaman syndicate space suit, they won't be able to go stealth if it is decided that the nukeops want to go stealth.

At the end of the day there's no real benefits (other than "I get to play the species I want yay") to this and only hampers the nuke ops team as a whole.

File: code/game/gamemodes/nuclear/nuclear.dm [hunk 1]
```diff
@@ -242,38 +220,105 @@ proc/issyndicate(mob/living/M as mob)
...
+
+	synd_mob.rejuvenate() //cancel any damage taken while processing game start
 	var/obj/item/implant/explosive/E = new/obj/item/implant/explosive(synd_mob)
 	E.implant(synd_mob)
 	synd_mob.faction |= "syndicate"
 	synd_mob.update_icons()
 	return 1
 
+/datum/game_mode/proc/choose_syndicate_species(datum/mind/synd_mind, var/leader = FALSE, var/agent_number = 1)
+	spawn(0) //prevent roundstart delay
```
Comment (Fox-McCloud) [inline code/game/gamemodes/nuclear/nuclear.dm]: That's not how a spawn works.

**Comment by datlo** [comment]
Literally all of these arguments can be used to remove species that aren't human from the game entirely... replace plasmaman syndicate suit by not being able to disguise as an antag.

This is also why I allow selecting to be a human if you wish to be one.

Most people care about and like to play as their character, it's.. kinda why we're here, right? 

Incidentally, this also makes stealth ops more effective for all but plasmamen. Now you can be any race instead of "Johnny McHuman DefinitelyNotANukeOpson".

**Comment by Fox-McCloud** [comment]
Not a big fan of this. One of the reasons Wizard allows or your species to shine through, but not Nukies is that it's a singular experience (for the most part, ragin' mages excluded, obviously). With nuke ops, you're dependent on others to work with. While species customization would be nice, it introduces a non-unified balance problem and also generates extremely frustrating situations whereby major strategies are out and out barred purely because someone wanted to play a particular race.

If species selection were to be a thing, it would have to be from a very small pool of species that don't have many defining attributes or strengths/weaknesses (Skrell, Vulpkanin, Tajaran, and Unathi).

Other races generates problems for the team, forcing everyone to play around them (Diona, IPC), while others just give a straight up edge for the game mode (Kidan).

But yeah, I don't think this is a very good idea. The amount of frustrations generated by this, when this was still allowed, were fairly high; especially given the pressures and stress of playing with a small group of other guys who are highly dependent on you. I don't think re-introducing them would be good.

**Comment by Ty-Omaha** [comment]
It's not all about the 1 person wishing to play _their_ species

Fox outlined it quite well

**Comment by datlo** [comment]
Well I suppose at this point it's just a difference of opinion.

I really don't like the idea of blacklisting certain species from being able to play as ops and not others, and since it seems this PR is controversial with staff, I'll close this for now.

At least I learned about gamemode roundstart setups and specie handling.

I'm going to open another PR to fix plasmamen burning up while the round setup.

**Comment by FalseIncarnate** [comment]
For reference, the change to make nuke ops human-only was bundled in https://github.com/ParadiseSS13/Paradise/pull/331

The title doesn't make it obvious it touched nuke ops, which is probably why you missed it @datlo plus it doesn't really have any comments on the nuke ops change specifically. That discussion was largely internal and didn't make its way to github

**Pull request closed without merging**